src/metatensor/models/alchemical_model/train.py

```diff
@@ -0,0 +1,268 @@
+    hypers: Dict = DEFAULT_HYPERS,
+    continue_from: Optional[str] = None,
+    output_dir: str = ".",
+    device_str: str = "cpu",
+):
+    if continue_from is None:
+        model = Model(
+            capabilities=requested_capabilities,
+            hypers=hypers["model"],
+        )
+        new_capabilities = requested_capabilities
+    else:
+        model = load_model(continue_from)
+        filtered_new_dict = {k: v for k, v in hypers["model"].items() if k != "restart"}
+        filtered_old_dict = {k: v for k, v in model.hypers.items() if k != "restart"}
+        if filtered_new_dict != filtered_old_dict:
+            logger.warn(
+                "The hyperparameters of the model have changed since the last "
+                "training run. The new hyperparameters will be discarded."
+            )
+        # merge the model's capabilities with the requested capabilities
+        merged_capabilities, new_capabilities = merge_capabilities(
+            model.capabilities, requested_capabilities
+        )
+        model.capabilities = merged_capabilities
+        # make the new model capable of handling the new outputs
+        for output_name in new_capabilities.outputs.keys():
+            model.add_output(output_name)
+
+    model_capabilities = model.capabilities
+
+    # Perform canonical checks on the datasets:
+    logger.info("Checking datasets for consistency")
+    check_datasets(
+        train_datasets,
+        validation_datasets,
+        model_capabilities,
+    )
+
+    # Calculating the neighbolists
+    logger.info("Checking and updating the neighbolists")
+    requested_neighbors_lists = model.requested_neighbors_lists()
+    check_and_update_neighbors_lists(
+        train_datasets + validation_datasets, requested_neighbors_lists
+    )
+
+    # Calculate the average number of atoms and neighbors in the training datasets:
+    average_number_of_atoms = get_average_number_of_atoms(train_datasets)
+    average_number_of_neighbors = get_average_number_of_neighbors(train_datasets)
+
+    # Given that currently multiple datasets are not supported, we can assume that:
+    average_number_of_atoms = average_number_of_atoms[0]
+    average_number_of_neighbors = average_number_of_neighbors[0]
+
+    # Set the normalization factor for the basis functions of the model:
+
+    # Set the normalization factors for the model:
+    model.set_normalization_factor(average_number_of_atoms)
+    model.set_basis_normalization_factor(average_number_of_neighbors)
+
+    logger.info(f"Training on device {device_str}")
+    if device_str == "gpu":
+        device_str = "cuda"
+    device = torch.device(device_str)
+    if device.type == "cuda":
+        if not torch.cuda.is_available():
+            raise ValueError("CUDA is not available on this machine.")
+        logger.info(
+            "A cuda device was requested. The neural network will be run on GPU, "
+            "but the SOAP features are calculated on CPU."
```

Comment:
is this true? I though this was a warning for rascaline-based models , but this is not the case here